firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/ModelFileDownloadService.java

```diff
@@ -301,16 +352,20 @@ public File loadNewlyDownloadedModelFile(CustomModel model) throws FirebaseMlExc
       if (fileDescriptor == null) {
         // reset original model - removing download id.
         sharedPreferencesUtil.setFailedUploadedCustomModelDetails(model.getName());
-        // todo call the download register?
+        removeOrCancelDownload(model.getName(), model.getDownloadId());
+        // todo log this?
         return null;
       }
 
       // Try to move it to destination folder.
-      File newModelFile = fileManager.moveModelToDestinationFolder(model, fileDescriptor);
+      File newModelFile;
+      try {
+        newModelFile = fileManager.moveModelToDestinationFolder(model, fileDescriptor);
+      } finally {
```

Comment:
If the move fails, or succeeds, we remove it. Would it be useful to know when it was removed because of a failure?

Reply:
yes - I have a major logging PR coming next week that will add lots of logging. Just got the Firelog PR in this week and need to get through my backlog first.